pkg/daemon/daemon.go

```diff
@@ -1475,20 +1498,44 @@ func (dn *Daemon) removeUpdateInProgressTaint(ctx context.Context) error {
 }
 
 func (dn *Daemon) shouldUpdateLayered() (*CoreOSDaemon, error) {
-	_, ok := dn.node.Annotations[constants.DesiredImageConfigAnnotationKey]
-	if ok {
+	if dn.isDesiredConfigAnImage() {
 		// sanity check that we're on CoreOS. The controller checks if a node runs CoreOS before setting desired image
 		if dn.os.IsCoreOSVariant() {
+			glog.Infof("Node is capable of image update")
 			return &CoreOSDaemon{dn}, nil
 		}
+
 		return nil, fmt.Errorf("can't treat a non-CoreOS node as layered")
 	}
 	return nil, nil
 }
+func (dn *Daemon) isDesiredConfigAnImage() bool {
+
+	if desiredConfigName, ok := dn.node.Annotations[constants.DesiredMachineConfigAnnotationKey]; ok {
+		// TODO(jkyros): This works for now, but we should probably add another annotation or at least
+		// do something more elegant than string prefix parsing
+		if strings.HasPrefix(desiredConfigName, "sha256:") {
```

Comment:
Preference has been expressed that we have separate config/image annotations and we clean up the ones we aren't using when we switch.

Reply:
I tried this with the most recent push, but it seems a little more invasive than I'd have liked due to some of the assumptions in the node controller's status (see status.go comment below)